### DISCUSSION TITLE
Why is IINA's Youtube playback so bad?

### DISCUSSION THREAD

**Discussion by psmpm** [iina/iina#5290] [General]
IINA's Youtube playback is incredibly painful to use, which is sad because the other option for native PiP is playing the video in Safari which has ads. Specifically, it takes forever to load for the first time and if you ever try to skip ahead it loads forever. Manually playing video/audio links in IINA grabbed through `yt-dlp -g` is much smoother, although they are separate links for audio and video and IINA doesn't have a way to combine media sources. What is IINA trying to do that is so unusably slow?

**Comment by low-batt** [comment]
For audio/video and subtitle rendering support IINA uses a library from the [mpv](https://mpv.io/) project. When IINA is given a YouTube URL it asks `libmpv` to open and play the video. It is mpv that makes use of [yt-dlp](https://github.com/yt-dlp/yt-dlp). The usual approach for such problems is to reproduce them with the mpv player and report them to the mpv project.
